### DISCUSSION TITLE
How can I add extra links to the top-right user dropdown

### DISCUSSION THREAD

**Discussion by vesper8** [laravel/nova-issues#4956] [Q&A] [answered]
Previously in Nova 3 this was as simple as publishing the views and editing `resources/views/vendor/nova/partials/user.blade.php`

How can I do that in Nova 4?

I need to add a link to the 2FA Configuration for a client.

Many thanks

**Comment by juse-less** [comment, marked as answer]
I'd recommend reading [this section](https://nova.laravel.com/docs/4.0/customization/menus.html#customizing-the-user-menu) in the documentation, which has some useful information on what works and what doesn't.

A small example, based on what I do myself:
```php
// NovaServiceProvider
public function boot(): void
{
    Nova::userMenu(function (\Illuminate\Http\Request $request, \Laravel\Nova\Menu\Menu $menu): \Laravel\Nova\Menu\Menu {
        // You can create and return a fresh Menu, if you want to ensure existing items are deleted.
        // Menu::make()
        // The Logout link will always be present at the bottom, from what I can tell, though.
        return $menu->append([
            MenuItem::link(\__('2FA Configuration'), \route('my.2fa.route')),
            // ...
        ]);
    });
}
```
